lib/base-command.js

```diff
@@ -142,6 +143,24 @@ class BaseCommand {
     return this.exec(args)
   }
 
+  // Compare the number of entries with what was expected
+  checkExpected (entries) {
+    if (!this.npm.config.isDefault('expect-results')) {
+      const expected = this.npm.config.get('expect-results')
+      if (!!entries !== !!expected) {
+        log.warn(this.name, `Expected ${expected ? '' : 'no '}results, got ${entries}`)
+        process.exitCode = 1
+      }
+    } else if (!this.npm.config.isDefault('expect-result-count')) {
+      const expected = this.npm.config.get('expect-result-count')
+      if (expected !== entries) {
+        /* eslint-disable-next-line max-len */
+        log.warn(this.name, `Expected ${expected} result${expected === 1 ? '' : 's'}, got ${entries}`)
+        process.exitCode = 1
+      }
+    }
+  }
+
```

Comment:
If this method is only used in `commands/query` should it be moved to that class until we have a need for it in more places? My concern is that previously methods added to the BaseClass were difficult to remove/refactor as we had to account for all it's usage.

Reply:
The intent is for almost EVERYTHING to be able to use this. `npm audit` `npm outdated` etc.

Reply:
So in this one isolated case we have prematurely optimized for the intended use of this.